lib/utils.js

```diff
@@ -13,16 +15,19 @@ const internals = {};
 
 module.exports.log = function (err, options, callback) {
 
-    const log = Fs.createWriteStream(options.logPath, options.writeStreamOptions);
-    const formattedErr = {
-        message: err.message,
-        stack: err.stack,
-        timestamp: Date.now()
-    };
+    Mkdirp(Path.dirname(options.logPath), () => {
```

Comment:
I've never used mocks before so bear with me. It sounds like you are telling me to 

1. create a new "fake" `Mkdirp` function that does nothing but call the callback with an Error
2. **Somehow** inject the reference to the new fake function into `PoopUtils.log` thereby replacing the "real" `Mkdirp` function
3. call `PoopUtils.log` and assert that the callback is called with an error (which confirms that the "fake" function was called and not the "real" function.

What I don't understand is how to inject the "fake" `Mkdirp` into `PoopUtils.log`. Help.